File: solutions/java/calculator-conundrum/2/src/main/java/CalculatorConundrum.java
class CalculatorConundrum {
    public static String calculate(int operand1, int operand2, String operation) {
        return switch (operation) {
            case null ->
                throw new IllegalArgumentException("Operation cannot be null");
            
            case "" ->
                throw new IllegalArgumentException("Operation cannot be empty");
            
            case "+" -> 
                Integer.valueOf(operand1) + " + "
                + Integer.valueOf(operand2) + " = "
                + Integer.valueOf(operand1 + operand2);
                
            case "*" ->
                Integer.valueOf(operand1) + " * "
                + Integer.valueOf(operand2) + " = "
                + Integer.valueOf(operand1 * operand2);

            case "/" -> {
                try {
                    yield Integer.valueOf(operand1) + " / "
                        + Integer.valueOf(operand2) + " = "
                        + Integer.valueOf(operand1 / operand2);
                } catch (ArithmeticException ex) {
                    throw new IllegalOperationException(
                        "Division by zero is not allowed", ex
                    );
                }
            }
                
            default ->
                throw new IllegalOperationException(
                    "Operation '" + operation + "' does not exist"
                );
        };
    }
}

File: solutions/csharp/calculator-conundrum/1/CalculatorConundrum.cs
public static class SimpleCalculator
{
    public static string Calculate(int operand1, int operand2, string? operation) => operation switch
    {
        null => throw new ArgumentNullException("Operation cannot be null"),
        "" => throw new ArgumentException("Operation cannot be empty"),
        "+" => $"{operand1} + {operand2} = {operand1 + operand2}",
        "*" => $"{operand1} * {operand2} = {operand1 * operand2}",
        "/" => Divide(operand1, operand2),
        _ => throw new ArgumentOutOfRangeException($"Operation {operation} unsupported"),
    };

    private static string Divide(int operand1, int operand2)
    {
        try
        {
            return $"{operand1} / {operand2} = {operand1 / operand2}";
        }
        catch (DivideByZeroException)
        {
            return "Division by zero is not allowed.";
        }
    }
}
